include/structs/sprite_oam.h

```diff
@@ -108,10 +108,199 @@ typedef struct SpriteOAM
+}
+
+#define SpriteSetTileNum(spritePtr, _tileNum) \
+{ \
+    u32 _tileNumVal = _tileNum; \
+    _tileNumVal &= SPRITEOAM_MAX_TILENUM; \
+    _tileNumVal <<= SPRITEOAM_SHIFT_TILENUM; \
+    (spritePtr)->attrib3 &= ~SPRITEOAM_MASK_TILENUM; \
+    (spritePtr)->attrib3 |= _tileNumVal; \
+}
+
+#define SpriteSetPriority(spritePtr, _priority) \
+{ \
+    u32 _priorityVal = _priority; \
+    _priorityVal &= SPRITEOAM_MAX_PRIORITY; \
+    _priorityVal <<= SPRITEOAM_SHIFT_PRIORITY; \
+    (spritePtr)->attrib3 &= ~SPRITEOAM_MASK_PRIORITY; \
+    (spritePtr)->attrib3 |= _priorityVal; \
+}
+
+#define SpriteSetPalNum(spritePtr, _palNum)                \
+{                                                       \
+    u32 _palNumVal = _palNum;                               \
+    _palNumVal &= SPRITEOAM_MAX_PALETTENUM;                \
+    _palNumVal <<= SPRITEOAM_SHIFT_PALETTENUM;                \
+    (spritePtr)->attrib3 &= ~SPRITEOAM_MASK_PALETTENUM;            \
+    (spritePtr)->attrib3 |= _palNumVal;                        \
+}
+
+#define SpriteSetMatrixNum(spritePtr, _matrixNum)                \
+{                                                       \
+    u32 _matrixNumVal = _matrixNum;                               \
+    _matrixNumVal &= SPRITEOAM_MAX_MATRIXNUM;                \
+    _matrixNumVal <<= SPRITEOAM_SHIFT_MATRIXNUM;                \
+    (spritePtr)->attrib2 &= ~SPRITEOAM_MASK_MATRIXNUM;            \
+    (spritePtr)->attrib2 |= _matrixNumVal;                        \
+}
+
+// Needed for TryCreateModeArrows. No difference to SpriteSetMatrixNum other than not creating one additional local variable.
+#define SpriteSetMatrixNum_UseLocalVar(spritePtr, _matrixNum)                \
+{                                                       \
+    _matrixNum &= SPRITEOAM_MAX_MATRIXNUM;                \
+    _matrixNum <<= SPRITEOAM_SHIFT_MATRIXNUM;                \
+    (spritePtr)->attrib2 &= ~SPRITEOAM_MASK_MATRIXNUM;            \
+    (spritePtr)->attrib2 |= _matrixNum;                        \
+}
+
+#define SpriteSetSize(spritePtr, _size)                \
+{                                                       \
+    u32 _sizeVal = _size;                               \
+    _sizeVal &= SPRITEOAM_MAX_SIZE;                \
+    _sizeVal <<= SPRITEOAM_SHIFT_SIZE;                \
+    (spritePtr)->attrib2 &= ~SPRITEOAM_MASK_SIZE;            \
+    (spritePtr)->attrib2 |= _sizeVal;                        \
+}
+
+#define SpriteSetX(spritePtr, _x)                \
+{               \
+    u32 _xSpriteVal = _x;               \
+    _xSpriteVal &= SPRITEOAM_MAX_X;               \
+    _xSpriteVal <<= SPRITEOAM_SHIFT_X;               \
+    (spritePtr)->attrib2 &= ~SPRITEOAM_MASK_X;               \
+    (spritePtr)->attrib2 |= _xSpriteVal;               \
+}
+
+// Hacky way of matching functions in menu_input.c AddMenuCursorSprite_, sub_801332C sub_8013470
+#define SpriteSetX_MatrixNumSize0(spritePtr, _x)                \
```

Comment:
Probably can name this one like "SetMatrixNuM" and the other funcs can be like "ModifyMatrixNum", "ModifyX" etc

Reply:
Not a fan of this, because these names wouldn't be clear enough. Especially for someone who is not familiar with these, it may be confusing why there are two sets, and why `SetX` modifies also `MatrixNum` and `X`